Automattic/vip-go-mu-plugins
# Update upload name to include timestamp if file already exists.

## Description

This PR changes the naming of uploaded files. If a file already exists, we will change the filename to the format `my-file-{timestamp}.ext`. Previously, WP would rename the file to `my-file-{next-available-number}.ext` which could cause big delays when the same file name was used often.

## Changelog Description

###  Modified Upload Filename Format

When a file exists, rename the file with appended timestamp rather than adding incremental digit.

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [ n/a] This change works and has been tested locally (or has an appropriate fallback).
- [X ] This change works and has been tested on a Go sandbox.
- [X ] This change has relevant unit tests (if applicable).
- [n/a ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [n/a ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test

Test in Sandbox
1. Check out PR.
1. Go to media library
1. Upload file
1. Upload file again.
1. Verify that first file is saved with original filename and second filename has timestamp added to original filename.
